core/node/da_clients/src/object_store.rs

```diff
@@ -84,3 +88,36 @@ impl DataAvailabilityClient for ObjectStoreDAClient {
         None
     }
 }
+
+/// Used as a wrapper for the pubdata to be stored in the GCS.
+#[derive(Debug)]
+pub struct StorablePubdata {
+    pub data: Vec<u8>,
+}
+
+impl StoredObject for StorablePubdata {
+    const BUCKET: Bucket = Bucket::DataAvailability;
+    type Key<'a> = L1BatchNumber;
+
+    fn encode_key(key: Self::Key<'_>) -> String {
+        format!("l1_batch_{key}_pubdata.gzip")
+    }
+
+    fn serialize(&self) -> Result<Vec<u8>, BoxedError> {
+        let mut encoder = GzEncoder::new(Vec::new(), Compression::default());
+        encoder.write_all(&self.data[..])?;
+        encoder.finish().map_err(From::from)
+    }
+
+    fn deserialize(bytes: Vec<u8>) -> Result<Self, BoxedError> {
```

Comment:
A snapshot test for this functionality would be appreciated, to make sure that we don't accidentally break the (de)serialization logic. You can look at example in `protobuf_interface/tests/job_serialization.rs`.

Reply:
not sure that you meant it to use the mock storage, but I wrote one that does

Reply:
No, I meant that you encode some data (doesn't necessarily have to be the actual blob), and store it to the file.
Then in the test, you load this file, deserialize it via `StoredObject::deserialize`, and 
a) make sure deserialization succeeds
b) it contains expected data.

Without such test, if serialization format changes (e.g. imagine that we changed `GzEncoder` to something else), we may not notice that it's a breaking change, but once deployed, the code won't be able to access any of the stored blobs.

Reply:
added